src/world/Units/Unit.Legacy.cpp

```diff
@@ -12237,21 +12223,20 @@ void Unit::EventHealthChangeSinceLastUpdate()
 {
     int pct = GetHealthPct();
     if (pct < 35)
-    {
-        uint32 toset = AURASTATE_FLAG_HEALTH35;
-        if (pct < 20)
-            toset |= AURASTATE_FLAG_HEALTH20;
-        else
-            RemoveFlag(UNIT_FIELD_AURASTATE, AURASTATE_FLAG_HEALTH20);
-        SetFlag(UNIT_FIELD_AURASTATE, toset);
-    }
+        addAuraState(AURASTATE_FLAG_HEALTH35);
     else
-        RemoveFlag(UNIT_FIELD_AURASTATE, AURASTATE_FLAG_HEALTH35 | AURASTATE_FLAG_HEALTH20);
+        removeAuraState(AURASTATE_FLAG_HEALTH35);
+
+    if (pct < 20)
+        addAuraState(AURASTATE_FLAG_HEALTH20);
+    else
+        removeAuraState(AURASTATE_FLAG_HEALTH20);
+
 #if VERSION_STRING >= WotLK
-    if (pct < 75)
-        RemoveFlag(UNIT_FIELD_AURASTATE, AURASTATE_FLAG_HEALTH75);
+    if (pct > 75)
```

Comment:
Is this change intentional?

Reply:
Yes, the health75 state was added in wotlk and is actually above 75% health and not below like other health related aurastates :)